framework/auth/core.py

```diff
@@ -605,26 +675,51 @@ def verify_claim_token(self, token, project_id):
         return record['token'] == token
 
     def confirm_email(self, token):
-        if self.verify_confirmation_token(token):
-            email = self.email_verifications[token]['email']
-            self.emails.append(email)
-            # Complete registration if primary email
-            if email.lower() == self.username.lower():
-                self.register(self.username)
-                self.date_confirmed = dt.datetime.utcnow()
-            # Revoke token
-            del self.email_verifications[token]
-            # Clear unclaimed records, so user's name shows up correctly on
-            # all projects
-            self.unclaimed_records = {}
-            self.save()
-            # Note: We must manually update search here because the fullname
-            # field has not changed
-            self.update_search()
-            self.update_search_nodes()
-            return True
-        else:
-            return False
+        """Confirm the email address associated with the token"""
+        email = self._get_unconfirmed_email_for_token(token)
+
+        # If this email is confirmed on another account, abort
+        if User.find(Q('emails', 'eq', email)).count() > 0:
```

Comment:
I would think we would want to stay with insensitive to stay away from those issues.  Why would the standard go with case sensitivity?